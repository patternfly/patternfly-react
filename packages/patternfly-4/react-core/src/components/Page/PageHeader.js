import React from 'react';
import PropTypes from 'prop-types';
import styles from '@patternfly/react-styles/css/components/Page/page';
import { css } from '@patternfly/react-styles';
import { BarsIcon } from '@patternfly/react-icons';
import { Button, ButtonVariant } from '../../components/Button';

const propTypes = {
  /** Additional classes added to the page header */
  className: PropTypes.string,
  /** Component to render the logo/brand (e.g. <Brand />) */
  logo: PropTypes.node,
  /** Additional props passed to the logo anchor container */
  logoProps: PropTypes.object,
  /** Component to render the toolbar (e.g. <Toolbar />) */
  toolbar: PropTypes.node,
  /** Component to render the avatar (e.g. <Avatar /> */
  avatar: PropTypes.node,
  /** Component to render navigation within the header (e.g. <Nav /> */
  topNav: PropTypes.node,
  /** True to show the nav toggle button (toggles side nav) */
  showNavToggle: PropTypes.bool,
  /** Callback function to handle the side nav toggle button, managed by the Page component if the Page isManagedSidebar prop is set to true */
  onNavToggle: PropTypes.func,
  /** True if the side nav is shown  */
  isNavOpen: PropTypes.bool,
  /** Callback function to handle the side nav toggle button */
  'aria-label': PropTypes.string,
  /** Additional props are spread to the container <header> */
  '': PropTypes.any // eslint-disable-line react/require-default-props
};

const defaultProps = {
  className: '',
  logo: null,
  logoProps: null,
  toolbar: null,
  avatar: null,
  topNav: null,
  showNavToggle: false,
  isNavOpen: true,
  onNavToggle: () => undefined,
  'aria-label': 'Global navigation'
};

const PageHeader = ({
  className,
  logo,
  logoProps,
  toolbar,
  avatar,
  topNav,
  isNavOpen,
  showNavToggle,
  onNavToggle,
  'aria-label': ariaLabel,
  ...props
}) => (
  <header role="banner" className={css(styles.pageHeader, className)} {...props}>
    {(showNavToggle || logo) && (
      <div className={css(styles.pageHeaderBrand)}>
        {showNavToggle && (
          <div className={css(styles.pageHeaderBrandToggle)}>
            <Button
              id="nav-toggle"
              onClick={onNavToggle}
              aria-label={ariaLabel}
              aria-controls="page-sidebar"
              aria-expanded={isNavOpen ? 'true' : 'false'}
              variant={ButtonVariant.plain}
            >
              <BarsIcon />
            </Button>
          </div>
        )}
        {logo && (
          <a className={css(styles.pageHeaderBrandLink)} {...logoProps}>
            {logo}
          </a>
        )}
      </div>
    )}
    {/* Hide for now until we have the context selector component */}
    {/* <div className={css(styles.pageHeaderSelector)}>
      pf-c-context-selector
    </div> */}
    {topNav && <div className={css(styles.pageHeaderNav)}>{topNav}</div>}
    {(toolbar || avatar) && (
      <div className={css(styles.pageHeaderTools)}>
        {toolbar}
        {avatar}
      </div>
    )}
  </header>
);

PageHeader.propTypes = propTypes;
PageHeader.defaultProps = defaultProps;

export default PageHeader;
