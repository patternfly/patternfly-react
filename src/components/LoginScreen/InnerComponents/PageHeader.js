import React from 'react';
import PropTypes from 'prop-types';

const PageHeader = props => (
  <header className="login-pf-page-header">
    <img
      className={`login-pf-brand ${props.logoClass}`}
      src={props.logoSrc}
      alt="logo"
    />
    <p>{props.pageDescription}</p>
  </header>
);

PageHeader.propTypes = {
  logoClass: PropTypes.string,
  logoSrc: PropTypes.string,
  pageDescription: PropTypes.string
};

PageHeader.defaultProps = {
  logoClass: null,
  logoSrc: null,
  pageDescription: null
};

export default PageHeader;
