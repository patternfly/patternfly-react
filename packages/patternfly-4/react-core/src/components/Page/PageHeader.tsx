import * as React from 'react';
import styles from '@patternfly/react-styles/css/components/Page/page';
import { css } from '@patternfly/react-styles';
import { BarsIcon } from '@patternfly/react-icons';
import { Button, ButtonVariant } from '../../components/Button';
import { PageContextConsumer } from './Page';

export interface PageHeaderProps extends React.HTMLProps<HTMLDivElement> {
  /** Additional classes added to the page header */
  className?: string;
  /** Component to render the logo/brand (e.g. <Brand />) */
  logo?: React.ReactNode;
  /** Additional props passed to the logo anchor container */
  logoProps?: object;
  /** Component to use to wrap the passed <logo> */
  logoComponent?: React.ReactNode;
  /** Component to render the toolbar (e.g. <Toolbar />) */
  toolbar?: React.ReactNode;
  /** Component to render the avatar (e.g. <Avatar /> */
  avatar?: React.ReactNode;
  /** Component to render navigation within the header (e.g. <Nav /> */
  topNav?: React.ReactNode;
  /** True to show the nav toggle button (toggles side nav) */
  showNavToggle?: boolean;
  /** True if the side nav is shown  */
  isNavOpen?: boolean;
  /**
   * If true, manages the sidebar open/close state and there is no need to pass the isNavOpen boolean into
   * the sidebar component or add a callback onNavToggle function into the PageHeader component
   */
  isManagedSidebar?: boolean;
  /** Callback function to handle the side nav toggle button, managed by the Page component if the Page isManagedSidebar prop is set to true */
  onNavToggle?: () => void;
  /** Aria Label for the nav toggle button */
  'aria-label'?: string;
}

export const PageHeader = ({
  className = '',
  logo = null as React.ReactNode,
  logoProps = null as object,
  logoComponent = 'a',
  toolbar = null as React.ReactNode,
  avatar = null as React.ReactNode,
  topNav = null as React.ReactNode,
  isNavOpen = true,
  showNavToggle = false,
  onNavToggle = () => undefined as any,
  'aria-label': ariaLabel = 'Global navigation',
  ...props
}: PageHeaderProps) => {
  const LogoComponent = logoComponent as any;
  return (
    <PageContextConsumer>
      {({ isManagedSidebar, onNavToggle: managedOnNavToggle, isNavOpen: managedIsNavOpen }: PageHeaderProps) => {
        const navToggle = isManagedSidebar ? managedOnNavToggle : onNavToggle;
        const navOpen = isManagedSidebar ? managedIsNavOpen : isNavOpen;

        return (
          <header role="banner" className={css(styles.pageHeader, className)} {...props}>
            {(showNavToggle || logo) && (
              <div className={css(styles.pageHeaderBrand)}>
                {showNavToggle && (
                  <div className={css(styles.pageHeaderBrandToggle)}>
                    <Button
                      id="nav-toggle"
                      onClick={navToggle}
                      aria-label={ariaLabel}
                      aria-controls="page-sidebar"
                      aria-expanded={navOpen ? 'true' : 'false'}
                      variant={ButtonVariant.plain}
                    >
                      <BarsIcon />
                    </Button>
                  </div>
                )}
                {logo && (
                  <LogoComponent className={css(styles.pageHeaderBrandLink)} {...logoProps}>
                    {logo}
                  </LogoComponent>
                )}
              </div>
            )}
            {/* Hide for now until we have the context selector component */}
            {/* <div className={css(styles.pageHeaderSelector)}>
            pf-c-context-selector
          </div> */}
            {topNav && <div className={css(styles.pageHeaderNav)}>{topNav}</div>}
            {(toolbar || avatar) && (
              <div className={css(styles.pageHeaderTools)}>
                {toolbar}
                {avatar}
              </div>
            )}
          </header>
        );
      }}
    </PageContextConsumer>
  );
};
