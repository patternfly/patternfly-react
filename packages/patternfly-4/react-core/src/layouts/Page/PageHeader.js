import React from 'react';
import styles from '@patternfly/patternfly-next/layouts/Page/page.css';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import { BarsIcon } from '@patternfly/react-icons';
import { Button, ButtonVariant } from '../../components/Button';
import { PageContext } from './Page';

const propTypes = {
  /** Additional classes added to the page header */
  className: PropTypes.string,
  /** Component to render the logo/brand (e.g. <Brand />) */
  logo: PropTypes.node,
  /** Component to render the toolbar (e.g. <Toolbar />) */
  toolbar: PropTypes.node,
  /** Component to render the avatar (e.g. <Avatar /> */
  avatar: PropTypes.node,
  /** Component to render navigation within the header (e.g. <Nav /> */
  topNav: PropTypes.node,
  /** True to show the nav toggle button (toggles side nav) */
  showNavToggle: PropTypes.bool,
  /** Callback function to handle the side nav toggle button */
  onNavToggle: PropTypes.func,
  /** header should have tall modifier */
  isTall: PropTypes.bool
};

const defaultProps = {
  className: '',
  logo: null,
  toolbar: null,
  avatar: null,
  topNav: null,
  showNavToggle: false,
  onNavToggle: () => undefined,
  isTall: false
};

/* Added temporary style as a workaround to make dropdowns work until fix is made (patternfly-next #780) */
const PageHeader = ({ className, logo, toolbar, avatar, topNav, showNavToggle, onNavToggle, isTall, ...props }) => (
  <PageContext.Consumer>
    {context => (
      <header
        role="banner"
        className={css(styles.pageHeader, ((context && context.isTall) || isTall) && styles.modifiers.tall, className)}
        {...props}
      >
        <div className={css(styles.pageHeaderBrand)}>
          {showNavToggle && (
            <div className={css(styles.pageHeaderBrandToggle)}>
              <Button
                id="nav-toggle"
                onClick={onNavToggle}
                aria-label="Toggle primary navigation"
                variant={ButtonVariant.plain}
              >
                <BarsIcon />
              </Button>
            </div>
          )}
          <a className={css(styles.pageHeaderBrandLink)}>{logo}</a>
        </div>
        {/* Hide for now until we have the context selector component */}
        {/* <div className={css(styles.pageHeaderSelector)}>
          pf-c-context-selector
        </div> */}
        {topNav && <div className={css(styles.pageHeaderNav)}>{topNav}</div>}
        <div className={css(styles.pageHeaderTools)}>
          {toolbar}
          {avatar}
        </div>
      </header>
    )}
  </PageContext.Consumer>
);

PageHeader.propTypes = propTypes;
PageHeader.defaultProps = defaultProps;

export default PageHeader;
