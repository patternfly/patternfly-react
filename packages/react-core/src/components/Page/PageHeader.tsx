/* eslint-disable no-console */
import * as React from 'react';
import styles from '@patternfly/react-styles/css/components/Page/page';
import { css } from '@patternfly/react-styles';
import BarsIcon from '@patternfly/react-icons/dist/js/icons/bars-icon';
import { Button, ButtonVariant } from '../../components/Button';
import { PageContextConsumer, PageContextProps } from './Page';

export interface PageHeaderProps extends React.HTMLProps<HTMLDivElement> {
  /** Additional classes added to the page header */
  className?: string;
  /** Component to render the logo/brand, use <Brand /> */
  logo?: React.ReactNode;
  /** Additional props passed to the logo anchor container */
  logoProps?: object;
  /** Component to use to wrap the passed <logo> */
  logoComponent?: React.ReactNode;
  /** Component to render the header tools, use <PageHeaderTools />  */
  headerTools?: React.ReactNode;
  /** Component to render navigation within the header, use <Nav /> */
  topNav?: React.ReactNode;
  /** True to show the nav toggle button (toggles side nav) */
  showNavToggle?: boolean;
  /** True if the side nav is shown  */
  isNavOpen?: boolean;
  /** This prop is no longer managed through PageHeader but in the Page component. */
  isManagedSidebar?: boolean;
  /** Sets the value for role on the <main> element */
  role?: string;
  /** Callback function to handle the side nav toggle button, managed by the Page component if the Page isManagedSidebar prop is set to true */
  onNavToggle?: () => void;
  /** Aria Label for the nav toggle button */
  'aria-label'?: string;
}

export const PageHeader: React.FunctionComponent<PageHeaderProps> = ({
  className = '',
  logo = null as React.ReactNode,
  logoProps = null as object,
  logoComponent = 'a',
  headerTools = null as React.ReactNode,
  topNav = null as React.ReactNode,
  isNavOpen = true,
  isManagedSidebar: deprecatedIsManagedSidebar = undefined,
  role = undefined as string,
  showNavToggle = false,
  onNavToggle = () => undefined as any,
  'aria-label': ariaLabel = 'Global navigation',
  'aria-controls': ariaControls = null,
  ...props
}: PageHeaderProps) => {
  const LogoComponent = logoComponent as any;
  if ([false, true].includes(deprecatedIsManagedSidebar)) {
    console.warn(
      'isManagedSidebar is deprecated in the PageHeader component. To make the sidebar toggle uncontrolled, pass this prop in the Page component'
    );
  }
  return (
    <PageContextConsumer>
      {({ isManagedSidebar, onNavToggle: managedOnNavToggle, isNavOpen: managedIsNavOpen }: PageContextProps) => {
        const navToggle = isManagedSidebar ? managedOnNavToggle : onNavToggle;
        const navOpen = isManagedSidebar ? managedIsNavOpen : isNavOpen;

        return (
          <header role={role} className={css(styles.pageHeader, className)} {...props}>
            {(showNavToggle || logo) && (
              <div className={css(styles.pageHeaderBrand)}>
                {showNavToggle && (
                  <div className={css(styles.pageHeaderBrandToggle)}>
                    <Button
                      id="nav-toggle"
                      onClick={navToggle}
                      aria-label={ariaLabel}
                      aria-controls={ariaControls}
                      aria-expanded={navOpen ? 'true' : 'false'}
                      variant={ButtonVariant.plain}
                    >
                      <BarsIcon />
                    </Button>
                  </div>
                )}
                {logo && (
                  <LogoComponent className={css(styles.pageHeaderBrandLink)} {...logoProps}>
                    {logo}
                  </LogoComponent>
                )}
              </div>
            )}
            {topNav && <div className={css(styles.pageHeaderNav)}>{topNav}</div>}
            {headerTools}
          </header>
        );
      }}
    </PageContextConsumer>
  );
};
PageHeader.displayName = 'PageHeader';
