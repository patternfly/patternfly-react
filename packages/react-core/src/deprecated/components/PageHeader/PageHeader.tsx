/* eslint-disable no-console */
import * as React from 'react';
import styles from '@patternfly/react-styles/css/components/Page/page';
import { css } from '@patternfly/react-styles';
import BarsIcon from '@patternfly/react-icons/dist/esm/icons/bars-icon';
import { Button, ButtonVariant } from '../../../components/Button';
import { PageContextConsumer, PageContextProps } from '../../../components/Page/PageContext';

export interface PageHeaderProps extends React.HTMLProps<HTMLDivElement> {
  /** Additional classes added to the page header */
  className?: string;
  /** Component to render the logo/brand, use <Brand /> */
  logo?: React.ReactNode;
  /** Additional props passed to the logo anchor container */
  logoProps?: any;
  /** Component to use to wrap the passed <logo> */
  logoComponent?: React.ReactNode;
  /** Component to render the header tools, use <PageHeaderTools />  */
  headerTools?: React.ReactNode;
  /** Component to render navigation within the header, use <Nav /> */
  topNav?: React.ReactNode;
  /** True to show the nav toggle button (toggles side nav) */
  showNavToggle?: boolean;
  /** Id for the nav toggle button */
  navToggleId?: string;
  /** True if the side nav is shown  */
  isNavOpen?: boolean;
  /** Sets the value for role on the <main> element */
  role?: string;
  /** Callback function to handle the side nav toggle button, managed by the Page component if the Page isManagedSidebar prop is set to true */
  onNavToggle?: (event: React.MouseEvent) => void;
  /** Aria Label for the nav toggle button */
  'aria-label'?: string;
}

export const PageHeader: React.FunctionComponent<PageHeaderProps> = ({
  className = '',
  logo = null as React.ReactNode,
  logoProps = null as any,
  logoComponent,
  headerTools = null as React.ReactNode,
  topNav = null as React.ReactNode,
  isNavOpen = true,
  role = undefined as string,
  showNavToggle = false,
  navToggleId = 'nav-toggle',
  onNavToggle = (_event: React.MouseEvent) => undefined as any,
  'aria-label': ariaLabel = 'Global navigation',
  'aria-controls': ariaControls = null,
  ...props
}: PageHeaderProps) => {
  let LogoComponent = logoComponent as any;
  if (!logoComponent) {
    if (logoProps?.href !== undefined) {
      LogoComponent = 'a';
    } else {
      LogoComponent = 'span';
    }
  }
  return (
    <PageContextConsumer>
      {({
        isManagedSidebar,
        onSidebarToggle: managedOnSidebarToggle,
        isSidebarOpen: managedIsSidebarOpen
      }: PageContextProps) => {
        const navToggle = isManagedSidebar ? managedOnSidebarToggle : onNavToggle;
        const navOpen = isManagedSidebar ? managedIsSidebarOpen : isNavOpen;

        return (
          <header role={role} className={css(styles.pageHeader, className)} {...props}>
            {(showNavToggle || logo) && (
              <div className={css(styles.pageHeaderBrand)}>
                {showNavToggle && (
                  <div className={css(styles.pageHeaderBrandToggle)}>
                    <Button
                      id={navToggleId}
                      onClick={navToggle}
                      aria-label={ariaLabel}
                      aria-controls={ariaControls}
                      aria-expanded={navOpen ? 'true' : 'false'}
                      variant={ButtonVariant.plain}
                    >
                      <BarsIcon />
                    </Button>
                  </div>
                )}
                {logo && (
                  <LogoComponent className={css(styles.pageHeaderBrandLink)} {...logoProps}>
                    {logo}
                  </LogoComponent>
                )}
              </div>
            )}
            {topNav && <div className={css(styles.pageHeaderNav)}>{topNav}</div>}
            {headerTools}
          </header>
        );
      }}
    </PageContextConsumer>
  );
};
PageHeader.displayName = 'PageHeader';
