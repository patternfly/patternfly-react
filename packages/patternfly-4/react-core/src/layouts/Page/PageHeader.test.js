import React from 'react';
import { shallow } from 'enzyme';
import PageHeader from './PageHeader';

test('Check page vertical layout example against snapshot', () => {
  const Header = (
    <PageHeader logo="Logo" toolbar="Toolbar" avatar=" | Avatar" onNavToggle={() => undefined} isCondensed />
  );
  const view = shallow(Header);
  expect(view).toMatchSnapshot();
});
