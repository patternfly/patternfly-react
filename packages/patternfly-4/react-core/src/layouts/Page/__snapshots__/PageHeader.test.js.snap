// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check page vertical layout example against snapshot 1`] = `
.pf-l-page__header-brand-link {
  display: none;
  align-items: center;
  min-height: 2.5rem;
}
.pf-l-page__header-brand {
  display: block;
  padding-left: 1rem;
}
.pf-l-page__header-tools {
  display: block;
  align-items: center;
  margin-right: 1rem;
  margin-left: auto;
}
.pf-l-page__header.pf-m-tall {
  display: flex;
  grid-area: header;
  z-index: 300;
  flex-wrap: wrap;
  align-items: flex-end;
  max-width: 100%;
  min-height: 6.875rem;
}

<PageHeader
  avatar=" | Avatar"
  className=""
  isTall={true}
  logo="Logo"
  logoProps={null}
  onNavToggle={[Function]}
  showNavToggle={false}
  toolbar="Toolbar"
  topNav={null}
>
  <Consumer>
    <header
      className="pf-l-page__header pf-m-tall"
      role="banner"
    >
      <div
        className="pf-l-page__header-brand"
      >
        <a
          className="pf-l-page__header-brand-link"
        >
          Logo
        </a>
      </div>
      <div
        className="pf-l-page__header-tools"
      >
        Toolbar
         | Avatar
      </div>
    </header>
  </Consumer>
</PageHeader>
`;
